### ISSUE TITLE
Enhancement: [no-unsafe-declaration-merging] dissallow `interface` declaration merging

### ISSUE THREAD

**Issue by AlexanderHott** [typescript-eslint/typescript-eslint#11297]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-unsafe-declaration-merging/

### Description

For most application code, `interface` declaration merging is probably an accidental name conflict instead of the intended usage. I propose to either change the default behavior of `no-unsafe-declaration-merging`, or add an option disallowing `interface`-`interface` declaration merging. 

Changing the default value might be a breaking change, so adding an option seems like a better way to move forward.

### Fail

```TypeScript
interface Foo {
  x: string;
}

interface Foo {
  y: number;
}
```

### Pass

```TypeScript
interface Foo {
  x: string;
  y: number;
}
```

**Comment by bradzacher** [comment]
There's already coverage for this in `no-redeclare`:
https://typescript-eslint.io/rules/no-redeclare/#ignoredeclarationmerge

[playground](https://typescript-eslint.io/play/#ts=5.8.2&showAST=es&fileType=.tsx&code=JYOwLgpgTgZghgYwgAgGIHt3IN4FgBQyyAHgFzIDOYUoA5gNwEC%2BBBoksiKGWehyAT3IgArgFsARtEb4W%2BAkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1ukQBNEy8AIY90YANrhsORNGgdokADRTpWadkiUA5px4ARAcOhD8lDkwCys7YjEAzIfGSJV2AL5uAulM-ugA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)
